libs/barista-components/sunburst-chart/src/sunburst-chart.ts

```diff
@@ -135,27 +139,64 @@ export class DtSunburstChart {
     DtSunburstChartNode[]
   > = new EventEmitter();
 
-  /** Template portal for the default overlay */
-  // tslint:disable-next-line: use-default-type-parameter no-any
-  private _portal: TemplatePortal<any> | null;
+  /** @internal Overlay template */
+  @ContentChild(DtSunburstChartOverlay, { static: false, read: TemplateRef })
+  _overlay: TemplateRef<DtSunburstChartTooltipData>;
 
   /** Reference to the open overlay. */
-  private _overlayRef: OverlayRef;
+  private _overlayRef: DtOverlayRef<DtSunburstChartTooltipData> | null;
+
+  /** @internal Destroy subject to clear subscriptions on component destroy. */
+  private readonly _destroy$ = new Subject<void>();
 
   constructor(
-    private _overlayService: Overlay,
-    private _viewContainerRef: ViewContainerRef,
+    private _overlayService: DtOverlay,
     private _platform: Platform,
     // TODO: remove this sanitizer when ivy is no longer opt out
     private _sanitizer: DomSanitizer,
-    private _elementRef?: ElementRef<HTMLElement>,
-    @Optional()
-    @Inject(DT_UI_TEST_CONFIG)
-    private _config?: DtUiTestConfiguration,
+    private _elementRef: ElementRef<HTMLElement>,
+    private _resizer: DtViewportResizer,
+    private _cdr: ChangeDetectorRef,
   ) {}
 
+  /** AfterContentInit hook */
+  ngAfterContentInit(): void {
+    this._updateDimensions();
+    this._resizer
+      .change()
+      .pipe(
+        // delay to postpone the reflow to the next change detection cycle
+        takeUntil(this._destroy$),
+        delay(0),
+      )
+      .subscribe(() => {
+        this._updateDimensions();
+        this._cdr.markForCheck();
```

Comment:
Our coding guidelines state to not abbreviate variable names if at all possible.
```suggestion
        this._changeDetectorRef.markForCheck();
```